crates/rolldown/src/module_loader/module_loader.rs

```diff
@@ -679,8 +679,7 @@ impl<'a> ModuleLoader<'a> {
     });
     // if `inline_dynamic_imports` is set to be true, here we should not put dynamic imports to entries
     if !self.options.inline_dynamic_imports {
-      let mut dynamic_import_entry_ids = dynamic_import_entry_ids.into_iter().collect::<Vec<_>>();
-      dynamic_import_entry_ids.sort_unstable_by_key(|(idx, _)| modules.get(*idx).stable_id());
+      let dynamic_import_entry_ids = dynamic_import_entry_ids.into_iter().collect::<Vec<_>>();
```

Comment:
We can't sort it here because, in partial scan mode, the `idx` may not exists in `modules`